src/repr/src/adt/date.rs

```diff
@@ -63,10 +63,10 @@ impl Date {
     pub const LOW_DAYS: i32 = -2451545; // 4714-11-24 BC
 
     /// Largest date support by Materialize. Although Postgres can go up to
-    /// 5874897-12-31, chrono is limited to December 31, 262143, which we mirror
+    /// 5874897-12-31, chrono is limited to December 31, 262142, which we mirror
     /// here so we can use chrono's formatting methods and have guaranteed safe
     /// conversions.
-    pub const HIGH_DAYS: i32 = 95_015_644;
+    pub const HIGH_DAYS: i32 = 95_015_279;
```

Comment:
This is a little scary because a date could exist in MZ today that is after the new limit, do you know what would happen in this case? probably a panic?

Reply:
Yes unfortunately a panic will likely occur if a date of this value already exists. I mentioned to @benesch yesterday, I'm hopeful that since the max limit was defined by chrono and not by postgres, it's unlikely that any of our customers used it as some sort of notional timestamp maximum